### ISSUE TITLE
Trap door inside shop?

### ISSUE THREAD

**Issue by renanbirck** [NetHack/NetHack#663]
A while ago, I had a game (ttyrec is [here](https://github.com/NetHack/NetHack/files/7954560/2022-01-27.22.38.25.ttyrec.gz), the dumplog is [here](https://www.hardfought.org/userdata/n/nabru/nethack/dumplog/1643323107.nh.html)) where a trapdoor was generated **inside** a shop.

This strikes me as unfair and possibly a bug. Could this probably be checked?

**Comment by copperwater** [comment]
This is in Minetown, right? I think @entrez had a potential explanation for this - something to do with the level being flagged "inaccessibles" (makes the level generator see if there are areas unreachable from the stairs and if so it generates ways out in those areas such as teleportation items or trap doors), but this runs too early, before the level generator knows where the stairs are, so it considers the entire map inaccessible and sticks a trapdoor potentially anywhere on it, which could be inside a shop.

**Comment by entrez** [comment]
Yeah, that's a good summary of what I think is going on.  I sent an email to the devteam about it the other day, around the time we were discussing it.

**Comment by renanbirck** [comment]
@copperwater yep, in Minetown.
